modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -131,19 +131,21 @@ const setupPurify = (settings: DomParserSettings, schema: Schema): DOMPurifyI =>
 
   // We use this to add new tags to the allow-list as we parse, if we notice that a tag has been banned but it's still in the schema
   purify.addHook('uponSanitizeElement', (ele, evt) => {
-    const element = SugarElement.fromDom(ele);
-    const tagName = evt.tagName;
     // Just leave non-elements such as text and comments up to dompurify
-    if (!SugarNode.isElement(element) || SugarNode.name(element) === 'body') {
+    const tagName = evt.tagName;
+    if (ele.nodeType !== NodeTypes.ELEMENT || tagName === 'body') {
```

Comment:
This saves us having to lookup the node name from the DOM and also avoids constructing the sugar wrapper if not needed.